### ISSUE TITLE
Instana cannot find process

### ISSUE THREAD

**Issue by chrisjcodes** [instana/nodejs-sensor#168]
Trying to add instana to our Node/Express API and we're getting the following log in our Instana Agent using

`require('@instana/collector')();`

Version 1.71.2

```
2019-08-27T13:08:40.420-0700 | INFO  | na-http-thread-4 | LoggerEndpoint                   | 51 - com.instana.agent - 1.1.505 | Node.js collector (pid: 8, reporting pid: 8): Transitioning from agentHostLookup to unannounced
2019-08-27T13:08:40.421-0700 | ERROR | na-http-thread-1 | Nodejs                           | 114 - com.instana.discovery-nodejs - 1.2.11 | Could not find the process NodejsProcess{pid='8', pidFromParentNS='false', name='node', arguments=[app.js], cpuSetFileContent='/ecs/prod-21/89a71c9121a24a0f879dfa0d215ae004/106e69fc7ec0d0bb24c2c80934020d2402608990586186837eece669e79b57c0
', fd='14', inode='socket:[34521786]'}.
2019-08-27T13:08:40.424-0700 | INFO  | na-http-thread-3 | LoggerEndpoint                   | 51 - com.instana.agent - 1.1.505 | Node.js collector (pid: 8, reporting pid: 8): Announce attempt failed: Announce to agent failed with status code 404. 
```

Not sure what is causing this issue, any ideas?

**Comment by basti1302** [comment]
Hi @chrisjcodes,

please use https://support.instana.com/hc/en-us/requests/new for inquiries like that. Actually, one of your colleagues already opened a ticket with a log snippet that contains this issue (Ticket #9359).

There I can see the following timeline of events in the log:

```
2019-08-28T13:33:27.558-0700 | INFO | na-http-thread-1 | LoggerEndpoint | 51 - com.instana.agent - 1.1.507 | Node.js collector (pid: 8, reporting pid: 8): Transitioning from agentHostLookup to unannounced 
2019-08-28T13:33:27.568-0700 | ERROR | na-http-thread-2 | Nodejs | 114 - com.instana.discovery-nodejs - 1.2.11 | Could not find the process NodejsProcess{pid='8', pidFromParentNS='false', name='node', arguments=[app.js], cpuSetFileContent='/ecs/{REDACTED} 
', fd='14', inode='socket:[30036766]'}. 
2019-08-28T13:33:27.571-0700 | INFO | na-http-thread-4 | LoggerEndpoint | 51 - com.instana.agent - 1.1.507 | Node.js collector (pid: 8, reporting pid: 8): Announce attempt failed: Announce to agent failed with status code 404. Will retry in 60000ms 
2019-08-28T13:33:34.601-0700 | INFO | 08-c1070c5bf75b) | Docker | 153 - com.instana.sensor-docker - 1.1.55 | Activated Sensor 
2019-08-28T13:33:35.381-0700 | INFO | heduler-thread-3 | AgentHttpServiceImpl | 51 - com.instana.agent - 1.1.507 | Listening on {REDACTED} 
2019-08-28T13:34:27.578-0700 | INFO | na-http-thread-4 | LoggerEndpoint | 51 - com.instana.agent - 1.1.507 | Node.js collector (pid: 8, reporting pid: 8): Overwriting pid for reporting purposes to: 19583 
2019-08-28T13:34:27.580-0700 | INFO | na-http-thread-2 | LoggerEndpoint | 51 - com.instana.agent - 1.1.507 | Node.js collector (pid: 8, reporting pid: 8): Changing pid to 19583 
2019-08-28T13:34:27.581-0700 | INFO | na-http-thread-3 | LoggerEndpoint | 51 - com.instana.agent - 1.1.507 | Node.js collector (pid: 8, reporting pid: 19583): Transitioning from unannounced to announced 
2019-08-28T13:34:34.726-0700 | INFO | ae-61f5e82ffdf9) | Nodejs | 154 - com.instana.sensor-nodejs - 1.2.11 | Activated Sensor 
2019-08-28T13:34:34.753-0700 | INFO | ae-61f5e82ffdf9) | Process | 155 - com.instana.sensor-process - 1.1.36 | Activated Sensor for PID 19583 
2019-08-28T13:34:34.777-0700 | INFO | ae-61f5e82ffdf9) | NodejsTrace | 156 - com.instana.sensor-nodejs-trace - 1.2.3 | Activated Sensor 
2019-08-28T13:34:37.591-0700 | INFO | na-http-thread-3 | LoggerEndpoint | 51 - com.instana.agent - 1.1.507 | Node.js collector (pid: 8, reporting pid: 19583): Reporting unhandled promise rejections is disabled. 
2019-08-28T13:34:37.591-0700 | INFO | na-http-thread-4 | LoggerEndpoint | 51 - com.instana.agent - 1.1.507 | Node.js collector (pid: 8, reporting pid: 19583): Agent is ready to accept. 
2019-08-28T13:34:37.591-0700 | INFO | na-http-thread-2 | LoggerEndpoint | 51 - com.instana.agent - 1.1.507 | Node.js collector (pid: 8, reporting pid: 19583): Transitioning from announced to agentready 
2019-08-28T13:34:37.593-0700 | INFO | na-http-thread-4 | LoggerEndpoint | 51 - com.instana.agent - 1.1.507 | Node.js collector (pid: 8, reporting pid: 19583): Reporting uncaught exceptions is disabled. 
2019-08-28T13:34:37.607-0700 | INFO | na-http-thread-3 | LoggerEndpoint | 51 - com.instana.agent - 1.1.507 | Node.js collector (pid: 8, reporting pid: 19583): The Instana Node.js collector is now fully initialized. 
```

TL;DR: The Instana agent needs to inspect the Docker container the Node.js app is running in to be able to match the process information the Node.js app is reporting during the initial handshake to an actual OS level process the agent sees (directly on the host or in a container). The first attempt failed because we had not yet inspected the Docker container. However, we retry this automatically. Once the Docker info was there, the handshake between the Node.js app and the Instana agent worked, as you can see.
